galata/test/jupyterlab/notebook-mermaid-diagrams.test.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { expect, galata, test } from '@jupyterlab/galata';
+import * as path from 'path';
+import { Locator } from '@playwright/test';
+
+const fileName = 'mermaid_diagrams.ipynb';
+
+const EXPECTED_MERMAID_ORDER = [
+  'flowchart',
+  'sequence',
+  'class',
+  'state',
+  'er',
+  'journey',
+  'gantt',
+  'pie',
+  'quadrant',
+  'requirement',
+  'c4',
+  'mindmap',
+  'timeline',
+  'sankey',
+  'xy'
+];
+
+/**
+ * Workaround for playwright not handling screenshots
+ * for elements larger than viewport, derived from:
+ * https://github.com/microsoft/playwright/issues/13486#issuecomment-1112012053
+ */
+async function resizePageAndScreenshot(locator: Locator) {
```

Comment:
Perhaps this is something that should be hoisted up outside of a deeply-nested `utils`?

Reply:
Yes, we could add it as a galata helper in the future if it turns out useful elsewhere too. For now it's easier to do it here because this way we don't need to worry about it becoming a public API yet ;)